pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -237,44 +240,84 @@ func (cm *certificateManager) getKeyPair(cli client.Client, secretName, secretNa
 		return nil, nil, err
 	}
 
+	// First, check that it's not expired.
 	if x509Cert.NotAfter.Before(time.Now()) || x509Cert.NotBefore.After(time.Now()) {
 		if !readCertOnly && strings.HasPrefix(x509Cert.Issuer.CommonName, rmeta.TigeraOperatorCAIssuerPrefix) {
 			if cm.keyPair.CertificateManagement != nil {
 				// When certificate management is enabled, we can simply return a certificate management key pair;
 				// the old secret will be deleted automatically.
 				return certificateManagementKeyPair(cm, secretName, nil), nil, nil
 			}
+
 			// We return nil, so a new secret will be created for expired (legacy) operator signed secrets.
 			return nil, nil, nil
 		}
-		// We return an error for byo secrets.
+
+		// The certificate was not issued by tigera-operator, or the caller indicated that certs shouldn't be modified.
+		// Consider this a BYO certificate, and error out rather than generate a new certificate.
 		return nil, nil, fmt.Errorf("secret %s is not valid at this date", secretName)
 	}
 	var issuer certificatemanagement.KeyPairInterface
 	if x509Cert.Issuer.CommonName == rmeta.TigeraOperatorCAIssuerPrefix {
```

Comment:
Do we need to do the `ValidForClientAndServer(x509Cert)` check for the else case of this? I'm thinking we need to do that for BYO certificates and let the user know if the provided certificates aren't sufficient.